dataframely/_serialization.py

```diff
@@ -118,3 +121,367 @@ def object_hook(self, dct: dict[str, Any]) -> Any:
                 )
             case _:
                 raise TypeError(f"Unknown type '{dct['__type__']}' in JSON data.")
+
+
+SerializedSchema = str
+SerializedCollection = str
+
+
+class StorageBackend(ABC):
```

Comment:
Potentially, we could think about creating a new subdirectory for this storage stuff; it's a lot of code for one file already 👀